strawberry_django_plus/utils/query.py

```diff
@@ -64,30 +74,37 @@ def filter_for_user_q(
     with_groups: bool = True,
     with_superuser: bool = False,
 ):
-    if with_superuser and user.is_active and user.is_superuser:
+    if with_superuser and user.is_active and getattr(user, "is_superuser", False):
         return qs
 
     if user.is_anonymous:
         return qs.none()
 
+    groups_field = None
+    try:
+        groups_field = cast("AbstractUser", user)._meta.get_field("groups")
+    except FieldDoesNotExist:
+        with_groups = False
+
     if isinstance(perms, str):
         perms = [perms]
 
-    model = qs.model
+    model = cast(Type[Model], qs.model)
     if model._meta.concrete_model:
-        model = model._meta.concrete_model
+        model = cast(Type[Model], model._meta.concrete_model)
 
     # We don't want to query the database here because this might not be async safe
     # Try to retrieve the ContentType from cache. If it is not there, we will
     # query it through the queryset
     ctype: Optional[ContentType] = None
-    try:
-        meta = model._meta
-        ctype = ContentType.objects._get_from_cache(meta)  # type: ignore
-    except KeyError:  # pragma:nocover
-        # If we are not running async, retrieve it
-        if not is_async():
-            ctype = ContentType.objects.get_for_model(model)
+    if has_contenttypes:
```

Comment:
In here you do this and the typing issues are solved :)

```suggestion
    if ContenType is not None:
```